adapter/PreciceInterface.c

```diff
@@ -47,6 +47,9 @@ void Precice_Setup( char * configFilename, char * participantName, SimulationDat
 	// Initialize coupling data
 	Precice_InitializeData( sim );
 
+	// find if coupling is implicit or explicit. Implicit coupling will return true at the very beginning and explicit will always return false
+	sim->coupling_explicit = !Precice_IsWriteCheckpointRequired();
```

Comment:
If you need such a boolean, it would be better to make is `coupling_implicit`. Usually we assume that implicit coupling is the special case that requires special treatment. You could then still write `if (!coupling_implicit)`.

But I think you don't need this check in the first place, it should be enough to call what you want always.

Reply:
sure, `!coupling_implicit` or `coupling_explicit` is no problem. But a check is needed. Because the call I want to happen always is only in the case of explicit coupling.